src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEMethodSymbol.cs

```diff
@@ -193,10 +193,16 @@ private UncommonFields CreateUncommonFields()
                 retVal._lazyObsoleteAttributeData = ObsoleteAttributeData.Uninitialized;
             }
 
-            if (!_packedFlags.IsUseSiteDiagnosticPopulated)
-            {
-                retVal._lazyUseSiteDiagnostic = CSDiagnosticInfo.EmptyErrorInfo; // Indicates unknown state.
-            }
+            //
+            // Do not set _lazyUseSiteDiagnostic !!!!
+            //
+            // "null" Indicates "no errors" or "unknown state".
+            // and we know which one of the states we have from IsUseSiteDiagnosticPopulated
+            //
+            // setting _lazyUseSiteDiagnostic to a sentinel value here would introduce
+            // a number of states with various permutations between IsUseSiteDiagnosticPopulated, UncommonFields and _lazyUseSiteDiagnostic
+            // some of them, in tight races, may lead to returning the sentinel as the diagnostics.
```

Comment:
The comments are mostly for the CR.
The whole pattern with sentinel+null+flag is problematic. It adds too many unnecessary states.

I will make a more comprehensive fix/cleanup in dev15.6
see:  [https://github.com/dotnet/roslyn/pull/20502](https://github.com/dotnet/roslyn/pull/20502) 

Here I just need to make the fix to unblock CoreFxLab, since this race causes build breaks.

---
In reply to: [124581343](https://github.com/dotnet/roslyn/pull/20497#discussion_r124581343) [](ancestors = 124581343)